kernel/src/hil/aes.rs

```diff
@@ -0,0 +1,14 @@
+use returncode::ReturnCode;
+
+pub trait AESDriver {
```

Comment:
We definitely will want `hash` functions, but that's likely a different interface, as is assymetric encryption, so yeah, I think `SymmetricEncryption` is good.